modules/47_regipol/regiCarbonPrice/postsolve.gms

```diff
@@ -195,7 +195,7 @@ pm_emiMktTarget_dev_iter(iteration, ttot,ttot2,ext_regi,emiMktExt) = pm_emiMktTa
 loop((ext_regi,ttot2)$regiANDperiodEmiMktTarget_47(ttot2,ext_regi),
   p47_targetConverged(ttot2,ext_regi) = 1;
   loop((ttot,emiMktExt,target_type_47,emi_type_47)$((pm_emiMktTarget(ttot,ttot2,ext_regi,emiMktExt,target_type_47,emi_type_47))),
-    if((abs(pm_emiMktTarget_dev(ttot,ttot2,ext_regi,emiMktExt)) > 0.01), !! if any emiMKt target did not converged
+    if((abs(pm_emiMktTarget_dev(ttot,ttot2,ext_regi,emiMktExt)) > 0.01), !! if emiMKt target did not converged
       p47_targetConverged(ttot2,ext_regi) = 0;
```

Comment:
not sure if I overlook something, but if the target is specific for emiMktExt shouldn't also the parameter storing whether or not it converged contain this dimension? and accordingly, also the target adjustment only happen for this emiMktExt?

Reply:
The `p47_targetConverged` parameter is reflecting if all targets set to a given year converged. If this happens I can move on to deal with the next period target instead in the iteration.

For example, if I set two parallel targets in 2030, one for the ETS and one for the ESR, if only the 2030 ETS target converged in this iteration, I still need to run in this iteration the rescaling tax algorithm to the 2030 prices, as the ESR target did not converged. 
I will only move on to the next year target (e.g. 2050), if all targets set to 2030 are converging in the given iteration.

Reply:
ok, and if one of the targets is already achieved, the value in `pm_factorRescaleemiMktCO2Tax(ttot,ttot2,ext_regi,emiMktExt)`  for this emiMktExt will be close to 1, so 
 `pm_taxemiMkt(ttot2,regi,emiMkt) = pm_taxemiMkt(ttot2,regi,emiMkt) * pm_factorRescaleemiMktCO2Tax(ttot,ttot2,ext_regi,emiMktExt);` won't change it anymore. 
 understood.

Reply:
yes, the value will be still updated in the ETS in the example I gave above, but as the deviation is small, the rescale factor will not fluctuate much from 1. 

In the corner case that the model has a discontinuity in this point and an small change to the ETS carbon price causes the ETS target to not converge anymore, the same code lines in the next iteration will tell that the model that ETS target needs to be updated for 2030. 

Only when all targets are converging in a given period that I move to the next one, 2050 in the example. 

Going further, if after updating the 2050 target in a previous iteration any of my previous targets do not converge anymore in the current iteration (e.g. 2030 ESR or ETS previously converged but not anymore because 2050 prices are lower), in this iteration I will keep the updated 2050 prices unchanged and will get back to solve the 2030 targets convergence assuming the new situation of decreased long term prices (2050 prices).